#!/usr/bin/env python
PACKAGE = "fence_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("a_channel_threshold", int_t, 0, "A* channel threshold", 170, 0, 255)
gen.add("b_channel_threshold", int_t, 0, "B* channel threshold", 170, 0, 255)
gen.add("x_edge_threshold", int_t, 0, "X gradient threshold", 150, 0, 255)
gen.add("y_edge_threshold", int_t, 0, "Y gradient threshold", 150, 0, 255)
gen.add("sum_threshold", int_t, 0, "Summed mask threshold", 85, 0, 255)
gen.add("dilate_iterations", int_t, 0, "", 14, 0, 30)
gen.add("erode_iterations", int_t, 0, "", 20, 0, 30)

gen.add("max_range", double_t, 0, "Clipping distance for fence points", 5.0, 0, 10.0)
gen.add("min_height", double_t, 0, "Ground clipping for fence points", 0.2, 0, 1.0)

gen.add("ransac_distance_threshold", double_t, 0, "Distance for inlier points", 0.1, 0, 1.0)

exit(gen.generate(PACKAGE, "fence_detection", "fence_detector_params"))
